torchx/components/dist.py

```diff
@@ -147,9 +157,13 @@ def ddp(
         raise ValueError("failed to compute role_name")
 
     rdzv_backend = "c10d"
-    if nnodes == 1:
+    if (not is_elastic and int(nnodes) == 1) or (
+        is_elastic and int(nnodes) == 0 and int(max_nnodes) == 1
```

Comment:
I don't think we need to support the zero node case. If you just ask for 1 node it'll be pending until resources are available regardless

Reply:
Done